Add Topiary tutorial resource

yannham opened the pull request:
Add the Yolo tree-sitter grammar and related artifacts that accompanies https://github.com/tweag/www/pull/1936/files.

yannham committed:
Add Topiary tutorial resource

Xophmeister commented on 2024-09-26-typescript-effect/src/index.ts:
You've got a bunch of commits from #27, which has been merged. I think you should be able to rebase off master and remove commits 3079391 and 003dd78 from your branch.

Xophmeister commented on TEMPORARY-topiary-tutorial-part-1/tree-sitter-yolo/Cargo.toml:
```diff
@@ -0,0 +1,27 @@
+readme = "README.md"
+keywords = ["incremental", "parsing", "tree-sitter", "yolo"]
+categories = ["parsing", "text-editors"]
+repository = "https://github.com/tree-sitter/tree-sitter-yolo"
```
Should this be `http://github.com/tweag/blog-resources`, or maybe just removed?

Xophmeister commented on TEMPORARY-topiary-tutorial-part-1/tree-sitter-yolo/Makefile:
```diff
@@ -0,0 +1,94 @@
+endif
+
+LANGUAGE_NAME := tree-sitter-yolo
+HOMEPAGE_URL := https://github.com/tree-sitter/tree-sitter-yolo
```
(Same comment RE repo URL)

Xophmeister commented on TEMPORARY-topiary-tutorial-part-1/tree-sitter-yolo/bindings/go/binding_test.go:
```diff
@@ -0,0 +1,15 @@
+	"testing"
+
+	tree_sitter "github.com/tree-sitter/go-tree-sitter"
+	tree_sitter_yolo "github.com/tree-sitter/tree-sitter-yolo/bindings/go"
```
(Same comment RE repo URL)

Xophmeister commented on TEMPORARY-topiary-tutorial-part-1/tree-sitter-yolo/go.mod:
```diff
@@ -0,0 +1,5 @@
+module github.com/tree-sitter/tree-sitter-yolo
```
(Ditto)

Xophmeister commented on TEMPORARY-topiary-tutorial-part-1/tree-sitter-yolo/package.json:
```diff
@@ -0,0 +1,56 @@
+  "name": "tree-sitter-yolo",
+  "version": "0.1.0",
+  "description": "A tree-sitter parser for the Yolo programming language",
+  "repository": "github:tree-sitter/tree-sitter-yolo",
```
(Ditto)

Xophmeister commented on TEMPORARY-topiary-tutorial-part-1/tree-sitter-yolo/pyproject.toml:
```diff
@@ -0,0 +1,30 @@
+readme = "README.md"
+
+[project.urls]
+Homepage = "https://github.com/tree-sitter/tree-sitter-yolo"
```
(Ditto)

Xophmeister commented on TEMPORARY-topiary-tutorial-part-1/tree-sitter-yolo/topiary-yolo.ncl:
```diff
@@ -0,0 +1,8 @@
+  languages = {
+    yolo = {
+      extensions = ["yolo"],
+      grammar.source.path = "/home/yago/Pro/Tweag/blog-posts/topiary-tutorial/tree-sitter-yolo/yolo.so",
```
Maybe it would be worth putting a comment in here or a dummy path (or both), so people following along can change this appropriately.

Xophmeister commented on TEMPORARY-topiary-tutorial-part-1/tree-sitter-yolo/tree-sitter.json:
```diff
@@ -0,0 +1,34 @@
+      }
+    ],
+    "links": {
+      "repository": "https://github.com/tree-sitter/tree-sitter-yolo"
```
(Ditto)

Xophmeister commented on TEMPORARY-topiary-tutorial-part-1/tree-sitter-yolo/tree-sitter.json:
```diff
@@ -0,0 +1,34 @@
+      "repository": "https://github.com/tree-sitter/tree-sitter-yolo"
+    }
+  },
+  "bindings": {
+    "c": true,
+    "go": true,
+    "node": true,
+    "python": true,
+    "rust": true,
+    "swift": true
+  }
```
Is it possible to just create Rust bindings, for the sake of your tutorial?

Xophmeister reviewed (changes requested):
The Tree-sitter repo URL, which I assume was generated, is set to an invalid URL. I've added comments where I've spotted these, but I might have missed some. I'm not sure what can be done about this, as I suspect some of the bindings (e.g., Go) make productive use of the URL, rather than it being just informational... _Maybe_ we just need the Rust bindings, for your tutorial, if that's possible.

Your query and the actual grammar look fine :+1:

yannham committed:
Remove TEMPORARY prefix from topiary tutorial resource

yannham committed:
Remove all bindings but C

yannham committed:
Remove Cargo.toml; remove references to (non-existent) tree-sitter repo

yannham committed:
Get rid of non existent tree-sitter repo in package.json

yannham commented:
@Xophmeister I got rid of the TEMPORARY prefix and removed all but the C bindings (I don't even think we particularly care about the Rust bindings either), and removed all references I could find to the non existent tree-sitter/tree-sitter-yolo repo.

Xophmeister commented on topiary-tutorial-part-1/tree-sitter-yolo/bindings/c/tree-sitter-yolo.pc.in:
```diff
@@ -0,0 +1,11 @@
+
+Name: tree-sitter-yolo
+Description: @PROJECT_DESCRIPTION@
+URL: @PROJECT_HOMEPAGE_URL@
```
I assume this just gets replaced with an empty string and that's fine. (Also, see L64 in your `Makefile`.)

yannham replied:
I guess? At any rate we just need the makefile to work (which I tested), I suspect we could even get rid of those packageconfig file. I assume it's not very important either way

Xophmeister commented on topiary-tutorial-part-1/tree-sitter-yolo/topiary-yolo.ncl:
```diff
@@ -0,0 +1,8 @@
+  languages = {
+    yolo = {
+      extensions = ["yolo"],
+      grammar.source.path = "/home/yago/Pro/Tweag/blog-posts/topiary-tutorial/tree-sitter-yolo/yolo.so",
```
How about:
```suggestion
      # Update this path appropriately
      grammar.source.path = "/path/to/blog-resources/topiary-tutorial-part-1/tree-sitter-yolo/yolo.so",
```

yannham replied:
Ah, better :sweat_smile: thanks

Xophmeister reviewed (approved):
Looks good :+1: Thanks, @yannham

yannham committed:
Update topiary-tutorial-part-1/tree-sitter-yolo/topiary-yolo.ncl

Co-authored-by: Christopher Harrison <Xophmeister@users.noreply.github.com>
